src/utils/ethers-utils.ts

```diff
@@ -1,8 +1,19 @@
 import type { TransactionReceipt } from '@ethersproject/abstract-provider/lib'
 import { ErrorCode } from '@ethersproject/logger'
 
-export type EthersError = Error & { code: ErrorCode; reason: string }
+// https://docs.ethers.io/v5/api/providers/types/#providers-TransactionResponse
+export enum EthersTxReplacedReason {
+  repriced = 'repriced',
+  cancelled = 'cancelled',
+  replaced = 'replaced',
+}
+
+export type EthersError = Error & { code: ErrorCode; reason: EthersTxReplacedReason; receipt: TransactionReceipt }
 
 export const didRevert = (receipt: TransactionReceipt): boolean => {
   return receipt.status === 0
 }
+
+export const didReprice = (error: EthersError): boolean => {
+  return error.code === ErrorCode.TRANSACTION_REPLACED && error.reason === EthersTxReplacedReason.repriced
```

Comment:
Why do you need to check both the code and the reason? Just the reason seems enough.

Reply:
I've updated the condition as per your comment.